src/index.ts

```diff
@@ -874,8 +875,16 @@ class ThreadPool {
   }
 
   _maybeDrain () {
-    if (this.taskQueue.size === 0 && this.skipQueue.length === 0) {
+    const totalCapacity = this.options.maxQueue + this.pendingCapacity();
+    const totalQueueSize = this.taskQueue.size + this.skipQueue.length;
+
+    if (totalQueueSize === 0) {
       this.publicInterface.emit('drain');
+      this.needsDrain = false;
+    }
+
+    if (totalQueueSize > totalCapacity) {
+      this.needsDrain = true;
```

Comment:
shouldn't it be an event emitter instead?

Reply:
The original issue seems to aim for a `property` so they can use it to make decisions further. And, you're right, an event can also provide good ergonomics so system metrics or further actions can be taken.

As args of the event, we can pass `totalCapacity`, and `totalQueueSize` at the moment the event takes place.

wdyt?

Reply:
Added in b2952f3, let me know what you thinkg